epsagon/trace.py

```diff
@@ -761,8 +768,47 @@ def remove_ignored_keys(self, input_dict):
                     if isinstance(value, dict):
                         self.remove_ignored_keys(value)
 
-    # pylint: disable=W0703
     def send_traces(self):
+        """
+        If trace size exceeds the maximum size, and split flag is on
+        then split the trace into multiple traces.
+        :return: None
+        """
+        if self.split_on_send and self.length > self._max_trace_size:
+            self._send_trace_split()
+        else:
+            self._send_traces()
+
+    def _send_trace_split(self):
```

Comment:
simplify this code by having an iterator that will return each time a batch of events that are <= trace size. and this way would also dumps only individual events instead of re-serializing the batch multiple time as it done here

for events_batch in get_next_events_batch(): events_batch.add(runner), increment fragment_id, send traces.

Reply:
You have to dump the events to JSON in order to understand whether it's size to too big.
It'll be fixed in a later version of our instrumentation code (as discussed with @ranrib ).